extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcCrudQueryGenerator.java

```diff
@@ -0,0 +1,241 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.polaris.extension.persistence.relational.jdbc.models.ModelGrantRecord;
+import org.apache.polaris.extension.persistence.relational.jdbc.models.ModelPrincipalAuthenticationData;
+
+public class JdbcCrudQueryGenerator {
+
+  private static final Pattern CAMEL_CASE_PATTERN =
+      Pattern.compile("(?<=[a-z0-9])[A-Z]|(?<=[A-Z])[A-Z](?=[a-z])");
+
+  public static String generateSelectQuery(
+      Class<?> entityClass, String filter, Integer limit, Integer offset, String orderBy) {
+    String tableName = getTableName(entityClass);
+    List<String> fields = new ArrayList<>();
+
+    for (Field field : entityClass.getDeclaredFields()) {
+      fields.add(camelToSnake(field.getName()));
+    }
+
+    String columns = String.join(", ", fields);
+    StringBuilder query =
+        new StringBuilder("SELECT ").append(columns).append(" FROM ").append(tableName);
+    if (filter != null && !filter.isEmpty()) {
+      query.append(" WHERE ").append(String.join(" AND ", filter));
+    }
+    return query.toString();
+  }
+
+  public static String generateSelectQuery(
+      Class<?> entityClass,
+      Map<String, Object> whereClause,
+      Integer limit,
+      Integer offset,
+      String orderBy) {
+    String tableName = getTableName(entityClass);
+    List<String> fields = new ArrayList<>();
+
+    for (Field field : entityClass.getDeclaredFields()) {
+      fields.add(camelToSnake(field.getName()));
+    }
+
+    String columns = String.join(", ", fields);
+    StringBuilder query =
+        new StringBuilder("SELECT ").append(columns).append(" FROM ").append(tableName);
+
+    if (whereClause != null && !whereClause.isEmpty()) {
+      query.append(generateWhereClause(whereClause));
+    }
+
+    if (orderBy != null && !orderBy.isEmpty()) {
+      query.append(" ORDER BY ").append(orderBy);
+    }
+
+    if (limit != null) {
+      query.append(" LIMIT ").append(limit);
+    }
+
+    if (offset != null && limit != null) { // Offset only makes sense with limit.
+      query.append(" OFFSET ").append(offset);
+    }
+
+    return query.toString();
+  }
+
+  public static String generateInsertQuery(Object object, Class<?> entityClass) {
+    String tableName = getTableName(entityClass);
+    if (object == null || tableName.isEmpty()) {
+      return null; // Or throw an exception
+    }
+
+    Class<?> objectClass = object.getClass();
+    Field[] fields = objectClass.getDeclaredFields();
+    List<String> columnNames = new ArrayList<>();
+    List<String> values = new ArrayList<>();
+
+    for (Field field : fields) {
+      field.setAccessible(true); // Allow access to private fields
+      try {
+        Object value = field.get(object);
+        if (value != null) { // Only include non-null fields
+          columnNames.add(camelToSnake(field.getName()));
```

Comment:
Would it not be preferable to delegate this to `ModelEntity` (or other value classes)? We have a hard dependency between classes and column names anyway, but proper delegation will avoid obscure reflection calls and make all dependencies explicit.

Reply:
can you please elaborate, are you suggesting along the lines of each Model having its own ResultSet to object converter ? and it has class property check, i personally prefer this mostly because rather than doing this : 
rs.getLong(<colum-idex>) or rs.getString(<colum_name>) we could have this in a loop without much concern.

Reply:
I mean rather than introspecting classes via java reflection, we could call a method like `Map<String, String> ModelEntity.sqlParameters()`

Reply:
I see, My thought process was, having an approach like reflection helps to not keep this mapping which essentially will be like all the members of the class, as our current approach is we essentially ask to get all the projected columns (its select *), which in this case is all the member.

 please let me know if you feel strongly about this considering above.